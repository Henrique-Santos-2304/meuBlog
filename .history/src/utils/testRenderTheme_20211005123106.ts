import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
