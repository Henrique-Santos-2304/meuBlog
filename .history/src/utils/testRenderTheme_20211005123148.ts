import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
