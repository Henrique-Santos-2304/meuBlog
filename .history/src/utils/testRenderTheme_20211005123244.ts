import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
