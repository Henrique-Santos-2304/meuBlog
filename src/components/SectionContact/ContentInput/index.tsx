import * as S from "./styles";

type props = {
  children: React.ReactNode;
  error: string | undefined;
  typeLabel: string;
  txtLabel: string;
};
const ContentInput = ({ children, error, typeLabel, txtLabel }: props) => (
  <S.ContentInputs>
    <S.Label htmlFor={typeLabel}>{txtLabel}</S.Label>
    {children}
    {error && (
      <S.Alert role="alert">
        <S.Icon></S.Icon>
        <p>{error}</p>
      </S.Alert>
    )}
  </S.ContentInputs>
);

export default ContentInput;
