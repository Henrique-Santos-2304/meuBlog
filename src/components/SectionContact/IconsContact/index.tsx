import * as S from "./styles";

const IconsContact = () => (
  <>
    <S.Wrapper>
      <S.Links href="https://github.com/Henrique-Santos-2304" passHref>
        <S.Anchor>
          <S.ContentIcon>
            <S.Github />
          </S.ContentIcon>
        </S.Anchor>
      </S.Links>

      <S.Links
        href="https://www.linkedin.com/in/henriquewebdeveloper/"
        passHref
      >
        <S.Anchor>
          <S.ContentIcon>
            <S.Linkedin />
          </S.ContentIcon>
        </S.Anchor>
      </S.Links>

      <S.Links href="https://spfc.net" passHref>
        <S.Anchor title="11.96636-5190">
          <S.ContentIcon>
            <S.Whatsapp />
          </S.ContentIcon>
        </S.Anchor>
      </S.Links>
    </S.Wrapper>
  </>
);

export default IconsContact;
