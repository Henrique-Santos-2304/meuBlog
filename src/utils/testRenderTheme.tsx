import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
