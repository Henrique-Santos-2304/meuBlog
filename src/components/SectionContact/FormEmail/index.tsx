/* eslint-disable @typescript-eslint/no-non-null-assertion */
import * as S from "./styles";
import { useEffect, useRef, useState } from "react";
import { useForm } from "react-hook-form";
import emailjs from "emailjs-com";
import ContentInput from "../ContentInput";

const FormEmail = () => {
  const {
    handleSubmit,
    register,
    formState: { errors },
  } = useForm();
  const formRef = useRef<HTMLFormElement>(null);
  const [msgForm, setMsgForm] = useState("");

  const onSubmit = async () => {
    await emailjs
      .sendForm(
        "service_ijjwv7k",
        "CONTACT_FORM",
        formRef.current!,
        "user_mB0pVxnJ1ebEVquIzWsUn"
      )
      .then(
        function () {
          setMsgForm("Enviado com sucesso");
        },
        function () {
          setMsgForm("Erro! Verifique seus dados");
        }
      );
  };
  useEffect(() => {
    setTimeout(() => {
      setMsgForm("");
    }, 10000);
  }, [msgForm]);

  return (
    <S.Wrapper>
      {msgForm && (
        <S.ModalForm aria-label="alerta de sucesso ou erro para envio de email">
          {msgForm}
        </S.ModalForm>
      )}
      <S.Form onSubmit={handleSubmit(onSubmit)} ref={formRef}>
        {/* Input de Nome */}
        <ContentInput
          error={errors.name && errors.name.message}
          typeLabel="name"
          txtLabel="Nome"
        >
          <S.Inputs
            id="name"
            type="text"
            placeholder="Digite seu nome"
            {...register("name", { required: "Digite seu nome" })}
          />
        </ContentInput>
        {/* Input de Email */}
        <ContentInput
          error={errors.email && errors.email.message}
          typeLabel="email"
          txtLabel="Email"
        >
          <S.Inputs
            id="email"
            type="email"
            placeholder="Digite seu email"
            {...register("email", {
              required: "Digite um email",
              pattern: {
                value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                message: "Formato de email invÃ¡lido",
              },
            })}
          />
        </ContentInput>

        {/* Input de Mensagem  */}
        <ContentInput
          error={errors.message && errors.message.message}
          typeLabel="message"
          txtLabel="Mensagem"
        >
          <S.TextAreas
            id="message"
            placeholder="Digite sua mensagem"
            {...register("message", { required: "Digite uma mensagem" })}
          />
        </ContentInput>

        <S.Button type="submit">Enviar</S.Button>
      </S.Form>
    </S.Wrapper>
  );
};

export default FormEmail;
