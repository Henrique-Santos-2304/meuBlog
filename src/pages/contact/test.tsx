import SectionContact from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<SectionContact />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<SectionContact />);
    expect(container).toMatchSnapshot();
  });
  it("should have all items the form ", () => {
    renderTheme(<SectionContact />);
    expect(
      screen.getByRole("heading", { name: /contato/i })
    ).toBeInTheDocument();
    expect(screen.getByText(/nome/i)).toBeInTheDocument();
    expect(screen.getByRole("textbox", { name: /nome/i })).toBeInTheDocument();
    expect(screen.getByText(/email/i)).toBeInTheDocument();
    expect(screen.getByRole("textbox", { name: /email/i })).toBeInTheDocument();
    expect(screen.getByText(/mensagem/i)).toBeInTheDocument();
    expect(
      screen.getByRole("textbox", {
        name: /campo de digitação de mensagem/i,
      })
    ).toBeInTheDocument();
  });
  it("should have link for whatsapp and button send message email", () => {
    renderTheme(<SectionContact />);
    expect(
      screen.getByRole("button", {
        name: /enviar/i,
      })
    ).toBeInTheDocument();
    expect(
      screen.getByLabelText(/Envio de mensagem no Whatsapp/i)
    ).toBeInTheDocument();
  });
});
