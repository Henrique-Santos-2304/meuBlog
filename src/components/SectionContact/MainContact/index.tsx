import Headings from "components/globalComponents/Headings";
import FormEmail from "../FormEmail";
import IconsContact from "../IconsContact";
import * as S from "./styles";

const MainContact = () => (
  <>
    <S.Wrapper id="contacts">
      <Headings
        msg="Contato"
        fontSizeText="2rem"
        marginText="0 0 0 3rem"
        aria-label="Titulo da seção"
      />
      <FormEmail />
    </S.Wrapper>
    <S.ContainerFooter>
      <IconsContact />
    </S.ContainerFooter>
  </>
);

export default MainContact;
