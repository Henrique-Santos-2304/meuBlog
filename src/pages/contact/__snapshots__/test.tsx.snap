// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContact /> should render  1`] = `
.c1 {
  position: absolute;
  bottom: 0.5rem;
  right: 0.5rem;
  padding: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  -webkit-transition: -webkit-transform 0.4s;
  -webkit-transition: transform 0.4s;
  transition: transform 0.4s;
  cursor: pointer;
}

.c1:hover {
  -webkit-transform: translate(-5%,-5%);
  -ms-transform: translate(-5%,-5%);
  transform: translate(-5%,-5%);
}

.c1 a svg {
  font-size: 4rem;
}

.c0 {
  padding: 3rem 0.5rem;
  width: 80%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
   
  <div
    class="chakra-container css-xawjhy"
  >
    <div
      class="chakra-container css-ed19g4"
    >
      <h2
        class="chakra-heading css-1voa61f"
      >
        Contato
      </h2>
      <div
        class="css-1k994a2"
      >
        <div
          class="css-1hnpd1e"
        >
          <form
            class="c0"
          >
            <div
              class="chakra-stack css-r2dqzy"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <label
                  class="chakra-form__label css-ohwb0c"
                  for="name"
                  id="field-1-label"
                >
                  Nome
                </label>
                <input
                  class="chakra-input css-13oby1y"
                  id="name"
                  name="name"
                  placeholder="Digite seu Nome"
                  type="text"
                />
              </div>
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <label
                  class="chakra-form__label css-ohwb0c"
                  for="email"
                  id="field-2-label"
                >
                  Email
                </label>
                <input
                  class="chakra-input css-13oby1y"
                  id="email"
                  name="email"
                  placeholder="Digite seu Email"
                  type="text"
                />
              </div>
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <label
                  class="chakra-form__label css-sw7jag"
                  for="message"
                  id="field-3-label"
                >
                  Mensagem
                </label>
                <textarea
                  aria-label="Campo de Digitação de Mensagem"
                  class="chakra-textarea css-1qcjni2"
                  id="field-3"
                  name="message"
                  placeholder="Digite sua Mensagem"
                  type="message"
                />
              </div>
              <button
                class="chakra-button css-1fvon0r"
                type="submit"
              >
                Enviar
                <span
                  class="chakra-button__icon css-1hzyiq5"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-icon css-onkibi"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </form>
        </div>
        <div
          class="c1"
        >
          <a
            aria-label="Envio de mensagem no Whatsapp"
            class="chakra-link css-njemur"
            href="https://api.whatsapp.com/send?1=pt_br&phone=5511966365190"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                11-96636.5190
              </title>
              <path
                d="M380.9 97.1C339 55.1 283.2 32 223.9 32c-122.4 0-222 99.6-222 222 0 39.1 10.2 77.3 29.6 111L0 480l117.7-30.9c32.4 17.7 68.9 27 106.1 27h.1c122.3 0 224.1-99.6 224.1-222 0-59.3-25.2-115-67.1-157zm-157 341.6c-33.2 0-65.7-8.9-94-25.7l-6.7-4-69.8 18.3L72 359.2l-4.4-7c-18.5-29.4-28.2-63.3-28.2-98.2 0-101.7 82.8-184.5 184.6-184.5 49.3 0 95.6 19.2 130.4 54.1 34.8 34.9 56.2 81.2 56.1 130.5 0 101.8-84.9 184.6-186.6 184.6zm101.2-138.2c-5.5-2.8-32.8-16.2-37.9-18-5.1-1.9-8.8-2.8-12.5 2.8-3.7 5.6-14.3 18-17.6 21.8-3.2 3.7-6.5 4.2-12 1.4-32.6-16.3-54-29.1-75.5-66-5.7-9.8 5.7-9.1 16.3-30.3 1.8-3.7.9-6.9-.5-9.7-1.4-2.8-12.5-30.1-17.1-41.2-4.5-10.8-9.1-9.3-12.5-9.5-3.2-.2-6.9-.2-10.6-.2-3.7 0-9.7 1.4-14.8 6.9-5.1 5.6-19.4 19-19.4 46.3 0 27.3 19.9 53.7 22.6 57.4 2.8 3.7 39.1 59.7 94.8 83.8 35.2 15.2 49 16.5 66.6 13.9 10.7-1.6 32.8-13.4 37.4-26.4 4.6-13 4.6-24.1 3.2-26.4-1.3-2.5-5-3.9-10.5-6.6z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
